asyncpg/connect_utils.py

```diff
@@ -448,44 +504,55 @@ def _parse_connect_arguments(*, dsn, host, port, user, password, passfile,
         resp = await reader.readexactly(1)
 
         if resp == b'S':
-            return sock.dup()
+            partial = functools.partial(
+                partial, ssl=ssl, server_hostname=server_hostname)
+        elif (ssl_is_advisory and
+                ssl.verify_mode == ssl_module.CERT_NONE and
```

Comment:
Given the logic above this is redundant.  Are there cases where `assert ssl_is_advisory and ssl.verify_mode == ssl_module.CERT_NONE` would fail?

Reply:
As the comment below describes, this is intentionally redundant to prevent possible future abuse of ssl_is_advisory.